### PR TITLE
feat: added typescript declarations

### PR DESCRIPTION
**Pull request by mateusz1913**
Follow up to https://github.com/vmurin/react-native-azure-auth/issues/84

### CODE REVIEW

File: types/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+ *
+ * One of Url options must be provided
+ *
+ * @param options
+ * @param options.authorityUrl
+ * @param options.baseUrl
+ * @param options.token
+ */
+declare class Client {
+  constructor(options: ClientOptions);
```
Comment (jotatoledo) [inline types/index.d.ts]: Seems like options has a default value, making it optional
https://github.com/vmurin/react-native-azure-auth/blob/v1.5.1/src/networking/index.js#L30

File: types/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+  patch(path: string, body: any): Promise<void>;
+  post(path: string, body: any): Promise<void>;
+  /**
+   * Helper function to send HTTP requests
+   *
+   * @param {String} method
+   * @param {String} url
+   * @param {Object} [body] - request body
+   */
+  request(method: string, url: string, body?: any): Promise<void>;
```
Comment (jotatoledo) [inline types/index.d.ts]: The type argument for the `Promise` is most definitely [not void](https://github.com/vmurin/react-native-azure-auth/blob/v1.5.1/src/networking/index.js#L91).
Considering the following interface def:

interface HttpResponse {
	status:string,
	ok: boolean,
    headers: Headers,
    blob?: Blob,
    json?: object,
    text?: string
}

```suggestion
  request(method: string, url: string, body?: any): Promise<HttpResponse>;
```

File: types/index.d.ts [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+   *
+   *  iOS: {YOUR_BUNDLE_IDENTIFIER}://${YOUR_BUNDLE_IDENTIFIER}/ios/callback
+   *
+   *  Android: {YOUR_APP_PACKAGE_NAME}://{YOUR_APP_PACKAGE_NAME}/android/callback
+   */
+  redirectUri?: string;
+}
+
+interface ClientOptions {
+  token: string;
```
Comment (jotatoledo) [inline types/index.d.ts]: Considering that https://github.com/vmurin/react-native-azure-auth/blob/v1.5.1/src/networking/index.js#L30 defines an empty object as default value for the options argument, it seems like all properties in this interface should be optional.

```suggestion
  token?: string;
```

File: types/index.d.ts (hunk 1 above)
Comment (vmurin) [reply types/index.d.ts]: @jotatoledo
options has an empty default value, but it's only used to make later destruction operator to work. Actually either `baseUrl` or `authorityUrl` are mandatory

File: types/index.d.ts (hunk 2 above)
Comment (vmurin) [reply types/index.d.ts]: Thanks! Very good suggestion!
@jotatoledo could you please add the interface as "_suggested change_" so I can take it over to PR
But I just would call it like `ClientResponse` as the `HttpResponse` is too common name and can interfere with something else and irritate ;).

File: types/index.d.ts (hunk 3 above)
Comment (vmurin) [reply types/index.d.ts]: `token` is optional anyway

**Pull request merged**